config/config.go

```diff
@@ -270,6 +274,52 @@ func (c *Config) Set(key string, value interface{}) {
 	c.onConfigChange()
```

Comment:
would it really be better to have `Set` take `value configTypes` instead of `value interface{}` too?
Doesn't still prevent setting wrong type though, so ambivalent about it.

Reply:
[We cannot have parametrized methods](https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#No-parameterized-methods). What you can do with generics is to have a generics function or a generics struct. The latter would mean inferring the `T configTypes` to the `Config` struct but then each `Config` instance can deal with one type at a time and that's not what we want.

Reply:
```go
type configTypes interface {
	int | int64
}

type Config[T configTypes] struct{}

func (c *Config[T]) Set(key string, v T) {
	// this is possible but then only one T per Config instance is allowed
}

func (c *Config) Set[T configTypes](key string, v T) {
	// this is NOT possible and it won't compile
}

func Set[T configTypes](key string, v T) {
	// this is perfectly fine
}
```

Reply:
Beside the technical difficulties to implement that, please keep in mind that `Set` is meant to be used in different ways due to the fact that we read env vars which are strings. For example this works:

```go
c := New()
c.Set("foo", "123ms")
t.Log(c.GetDuration("foo", 0, time.Millisecond)) // <- parsed correctly, it returns 123ms of type time.Duration

// ---

c := New()
c.Set("foo", "ciao")
t.Log(c.GetDuration("foo", 11, time.Millisecond)) // <- can't be parsed, it returns 11ms (default) with no error
```

Changing that would possibly lead to a bigger refactoring, outside the scope of this PR (and we might not even need it).

Reply:
>[We cannot have parametrized methods](https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#No-parameterized-methods)

TIL
thanks 👍

Reply:
> TIL thanks 👍

Me too, I had to Google `TIL` :joy: